pkg/config/tenancy/interceptor.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022 The Jaeger Authors.
```

Comment:
it's weird that we have interceptors and other functionality within package `config/*`. If config is only one of tenancy-related functions, then I would suggest:
```
pkg/
    tenancy/
        config.go (for TenancyConfig and validators)
        context.go (instead of being under storage/)
        grpc.go (for grpc-related pieces)
        http.go (for http-related pieces)
```